plugins/query-resource-limit-group/src/main/java/org/opensearch/plugin/resource_limit_group/TransportCreateResourceLimitGroupAction.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugin.resource_limit_group;
+
+import org.opensearch.action.support.ActionFilters;
+import org.opensearch.action.support.HandledTransportAction;
+import org.opensearch.cluster.metadata.ResourceLimitGroup;
+import org.opensearch.common.inject.Inject;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.plugin.resource_limit_group.service.Persistable;
+import org.opensearch.tasks.Task;
+import org.opensearch.threadpool.ThreadPool;
+import org.opensearch.transport.TransportService;
+
+/**
+ * Transport action for create Resource Limit Group
+ *
+ * @opensearch.internal
+ */
+public class TransportCreateResourceLimitGroupAction extends HandledTransportAction<
+    CreateResourceLimitGroupRequest,
+    CreateResourceLimitGroupResponse> {
+
+    private final ThreadPool threadPool;
+    private final Persistable<ResourceLimitGroup> resourceLimitGroupPersistenceService;
+
+    /**
+     * Constructor for TransportCreateResourceLimitGroupAction
+     *
+     * @param actionName - acrtion name
+     * @param transportService - a {@link TransportService} object
+     * @param actionFilters - a {@link ActionFilters} object
+     * @param threadPool - a {@link ThreadPool} object
+     * @param resourceLimitGroupPersistenceService - a {@link Persistable} object
+     */
+    @Inject
+    public TransportCreateResourceLimitGroupAction(
+        String actionName,
+        TransportService transportService,
+        ActionFilters actionFilters,
+        ThreadPool threadPool,
+        Persistable<ResourceLimitGroup> resourceLimitGroupPersistenceService
+    ) {
+        super(CreateResourceLimitGroupAction.NAME, transportService, actionFilters, CreateResourceLimitGroupRequest::new);
+        this.threadPool = threadPool;
+        this.resourceLimitGroupPersistenceService = resourceLimitGroupPersistenceService;
+    }
+
+    @Override
+    protected void doExecute(
+        Task task,
+        CreateResourceLimitGroupRequest request,
+        ActionListener<CreateResourceLimitGroupResponse> listener
+    ) {
+        ResourceLimitGroup resourceLimitGroup = new ResourceLimitGroup(
```

Comment:
Where are we adding validators for these fields ?

Reply:
When a request enters the rest class (eg. RestUpdateResourceLimitGroupAction), we call the function  UpdateResourceLimitGroupRequest.fromXContent(parser); to parse the new request. This function will call ResourceLimitGroup.fromXContentOptionalFields/ResourceLimitGroup.fromXContent, and within witch will have the validator for each resource limit group. As a result, all fields in UpdateResourceLimitGroupRequest already passed the validator in ResourceLimitGroup.